src/Components/WebAssembly/WebAssembly/src/Services/Logging/WebAssemblyConsoleLoggerProvider.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.JSInterop;
+
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Services
+{
+    /// <summary>
+    /// A provider of <see cref="WebAssemblyConsoleLogger{T}"/> instances.
+    /// </summary>
+    public class WebAssemblyConsoleLoggerProvider : ILoggerProvider
+    {
+        private ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>> _loggers;
+        private IJSInProcessRuntime _jsRuntime;
+        private bool _disposed;
+
+        /// <summary>
+        /// Creates an instance of <see cref="WebAssemblyConsoleLoggerProvider"/>.
+        /// </summary>
+        /// <param name="options">The options to create <see cref="WebAssemblyConsoleLogger"/> instances with.</param>
+        public WebAssemblyConsoleLoggerProvider(IJSInProcessRuntime jsRuntime)
+        {
+            _loggers = new ConcurrentDictionary<string, WebAssemblyConsoleLogger<object>>();
+            _jsRuntime = jsRuntime;
+        }
+
+        /// <inheritdoc />
+        public ILogger CreateLogger(string name)
+        {
+            return _loggers.GetOrAdd(name, loggerName => new WebAssemblyConsoleLogger<object>(name, _jsRuntime));
```

Comment:
Maybe this implementation comes from somewhere else, but I'm surprised it's not an error to call this with a `name` that's already registered.

Reply:
This is based on the implementation of the [ConsoleLoggerProvider](https://github.com/dotnet/extensions/blob/master/src/Logging/Logging.Console/src/ConsoleLoggerProvider.cs#L59) in Microsoft.Extensions.Logging. 

> since the underlying dictionary can't store more than one with the same name.

Hm. `GetOrAdd` is used here so that constraint should be satisfied even if the user calls `CreateLogger("someName")` twice.